e2e/test/scenarios/question/question-management.cy.spec.js

```diff
@@ -178,19 +180,100 @@ describe("managing question from the question's details sidebar", () => {
               cy.findByText("This question has been archived");
             });
 
-            it("should be able to add question to dashboard", () => {
-              openQuestionActions();
-              cy.findByTestId("add-to-dashboard-button").click();
+            describe("Add to Dashboard", () => {
+              it("should be able to add question to dashboard", () => {
+                openQuestionActions();
+                cy.findByTestId("add-to-dashboard-button").click();
+
+                modal().as("modal").findByText("Orders in a dashboard").click();
+
+                cy.get("@modal").should("not.exist");
+                // By default, the dashboard contains one question
+                // After we add a new one, we check there are two questions now
+                cy.get(".DashCard").should("have.length", 2);
+              });
+
+              it("should preselect the most recently visited dashboard", () => {
+                openQuestionActions();
+                cy.findByTestId("add-to-dashboard-button").click();
+
+                findSelectedItem().should("not.exist");
+
+                // before visiting the dashboard, we don't have any history
+                visitDashboard(1);
+
+                visitQuestion(ORDERS_QUESTION_ID);
+
+                openQuestionActions();
+                cy.findByTestId("add-to-dashboard-button").click();
+
+                findSelectedItem().should("have.text", "Orders in a dashboard");
+              });
+
+              onlyOn(user === "normal", () => {
+                it("should handle lost access", () => {
+                  cy.intercept(
+                    "GET",
+                    "/api/activity/most_recently_viewed_dashboard",
+                  ).as("mostRecentlyViewedDashboard");
+
+                  openQuestionActions();
+                  cy.findByTestId("add-to-dashboard-button").click();
+
+                  cy.wait("@mostRecentlyViewedDashboard").then(
```

Comment:
I'd rather not check for exact responses, maybe just for the endpoint being called (cy.wait)

Reply:
I wanted to test also BE responses

- visited - 200
- not visited - 204
- no permissions - 204